### ISSUE TITLE
Operations regarding characters are not working as intended

### ISSUE THREAD

**Issue by luc10921** [neo-project/neo-devpack-dotnet#681]
Using operations relating to characters is not working as intended, e.g.,:
```cs
	public static void Main()
	{
		string Str = "ΔδΣΦφ";
		int LengthOfStr = Str.Length;
		string SubStr1 = Str.Substring(1, 4);
		string SubStr2 = Str.Substring(2, 3);
		string SubStr3 = Str.Substring(3, 2);
		Console.WriteLine(Str);
		Console.WriteLine(LengthOfStr);
		Console.WriteLine(SubStr1);
		Console.WriteLine(SubStr2);
		Console.WriteLine(SubStr3);
	}
``` 
Running the code above in C# results in:
```
ΔδΣΦφ
5
δΣΦφ
ΣΦφ
Φφ
```
However, when running the same code in Neo, it ends up considering the number of bytes instead of the number of characters, so, every time a char that is not present in the ASCII table is used, the behaviour will diverge from C#.
![image](https://user-images.githubusercontent.com/49196318/136096915-5e55411c-44c3-4207-b9ac-22f29055e213.png)

**Comment by shargon** [comment]
It seems that it works as a byte array inside the VM

**Comment by lock9** [comment]
Yes @shargon , that is the issue, however, it is possible for the compiler to fix this, by checking the byte value

**Comment by shargon** [comment]
@lock9 how the compiler will distinguish between a string and a byte array without the type?
